pyuvsim/telescope.py

```diff
@@ -37,3 +41,278 @@ def __eq__(self, other):
+                for key in self.uvb_params.keys():
+                    val = getattr(bm, key)
+                    if key not in new_pars:
+                        new_pars[key] = []
+                    if val is not None:
+                        new_pars[key].append(val)
+        for key, vals in new_pars.items():
+            svals = set(vals)
+            if len(svals) == 1:
+                new_pars[key] = svals.pop()
+            elif len(svals) > 1:
+                try:
+                    raise ValueError('Conflicting settings for {}: {}'.format(key, str(svals)))
+                except ValueError as err:
+                    if strict:
+                        raise err
+                    else:
+                        warnings.warn(err.args[0])  # Use exception message as warning message.
+
+        new_pars = {k: v for k, v in new_pars.items() if not v == []}
+
+        return new_pars
+
+    def _set_params_on_uvbeams(self, beam_objs):
+        """
+        Set parameters on UVBeam objects using uvb_params.
+        """
+        for bm in np.atleast_1d(beam_objs):
+            if isinstance(bm, UVBeam):
+                for key, val in self.uvb_params.items():
+                    setattr(bm, key, val)
+
+    def __len__(self):
+        # Note that only one of these lists has nonzero length at a given time.
+        return len(self._obj_beam_list) + len(self._str_beam_list)
+
+    def __getitem__(self, ind):
+        if self.string_mode:
+            return self._str_beam_list[ind]
+        return self._obj_beam_list[ind]
+
+    def __setitem__(self, ind, value):
+        """
+        Insert into the beam list, converting objects to strings if in object mode,
+        or vice versa if in string mode.
+        """
+        if self.string_mode:
+            # If value is a string, then _scrape_uvb_params returns an empty dictionary.
+            newobj_params = self._scrape_uvb_params([value], strict=False)
+            mismatches = []
+            for key, val in newobj_params.items():
+                if key in self.uvb_params:
+                    if not val == self.uvb_params[key]:
+                        mismatches.append(key)
+            if len(mismatches) > 0:
+                compare = "\n\t".join([
+                    "current = {}, new = {}".format(self.uvb_params[key], newobj_params[key])
+                    for key in mismatches]
+                )
+                raise ValueError('UVBeam parameters do not match '
+                                 'those currently set: \n\t' + compare)
+
+            self._str_beam_list[ind] = self._obj_to_str(value)
+        else:
+            value = self._str_to_obj(value)
+            self._obj_beam_list[ind] = value
+            self._scrape_uvb_params(self._obj_beam_list, strict=False)
+
+    def append(self, value):
+        """
+        Append to the beam list, converting objects to strings if in object mode,
+        or vice versa if in string mode.
```

Comment:
This docstring will error with flake8 when we turn on docstring testing in Steven's PR -- The first line should be a single line sentence (ending in a period). More details should follow in text after a blank line.